### PR TITLE
[backport][widevine] add support for linux arm64

### PR DESCRIPTION
**Pull request by tmm1**
without these changes, the following error is observed:

    Initialize: Failed to load library: libwidevinecdm.so: undefined symbol: __aarch64_ldadd4_acq_rel
    Unable to load widevine shared library (libwidevinecdm.so)

cc #1128 https://github.com/emilsvennesson/script.module.inputstreamhelper/issues/530

cc @matthuisman @glennguy @peak3d @knaerzche @jernejsk

for reference:
- https://github.com/devins2518/zig/blob/545c3117ff2f88ee9f29414429a06cedebc0d8c0/lib/compiler_rt/aarch64_outline_atomics.zig#L1421-L1461
- https://github.com/gcc-mirror/gcc/blob/fe4608efc15b881ac908a3f90d7322736495ae72/libgcc/config/aarch64/lse-init.c
- https://github.com/gcc-mirror/gcc/blob/fe4608efc15b881ac908a3f90d7322736495ae72/libgcc/config/aarch64/lse.S
- https://godbolt.org/z/z8W7z1cqx

### CODE REVIEW

**Comment by CastagnaIT** [comment]
you can avoid ping peak3d it on threads is not on kodi team from many years
currently the only maintainers are @CastagnaIT and @glennguy

i think you have better knowledge of what is involved in loading the library where linux its not my area
so i do not have much to say

but i am a bit concerned about asm implementation for dynamic symbols
so how, or what we should do if something changes in the future? (maybe possible cases of different library versions?)
may this cause unhandled crashes on InputStreamAdaptive?
is this the only possible way?

i think its better ping also @phunkyfish

**Comment by tmm1** [comment]
I updated the code to remove the ASM. Technically the ASM version will have increased performance because it can use the LSE extensions, but I don't think that's a big deal here. I left that implementation in the git history for reference in case it becomes an issue in the future.

**Comment by CastagnaIT** [comment]
There is an additional problem to be taken in account
this implementation may be good for Nexus but for Omega we have plan to merge ssdwv project to the master project
there will no longer be an additional library builded, then `LoadDll` will be removed
I think you need to find an alternative

**Comment by tmm1** [comment]
Options would be:

1. build separate .so with just the atomic functions which can be dlopen
2. add atomic functions to kodi itself so they are always present in the main binary and available to use

**Comment by CastagnaIT** [comment]
@tmm1 can you create same PR for Omega branch?

**Comment by matthuisman** [comment]
So for arm64 Kodi devices, they will need minimum whatever version IA that is released after this is merged? If so, I'll add a IA version check in my addons to give error if IA version too old for 64bit

**Review by glennguy — APPROVED** [review]
Thanks @tmm1 for finding the arm64 libraries and kicking this off

**Comment by theofficialgman** [comment]
> Thanks @tmm1 for finding the arm64 libraries and kicking this off

where is my thanks 😆 , I am the one who noticed the widevine commit in chromium last fall, and that all the arm64 chromeOS CI was in place in January and asked in the google groups. But we really have Doug Anderson to thank.

**Comment by glennguy** [comment]
I guess I was trying to thank tmm1 for bringing it into the space here. Thanks to you for the digging in the first place. In the future though if there's new CDM platform/archs that become available and you find them please share the info with us so we know about it, we would be very grateful. 
If a tree falls in a forest...

**Comment by Schlumpf7** [comment]
so how i get widevine support to work on arm64? the inputstream adaptive helper still claims arm64 is not supported. 
How can i import it manually?

**Comment by glennguy** [comment]
@Schlumpf7 Have a read through this: https://github.com/emilsvennesson/script.module.inputstreamhelper/issues/530

**Comment by amazingfate** [comment]
Hello, there seems to be a gcc fix from fedora: https://bugzilla.redhat.com/show_bug.cgi?id=1830472
https://src.fedoraproject.org/rpms/gcc/c/95507e8b681b5013b87bbbcb0514dd27941e648a
Can I make use of this libwidevine library system wide?

**Pull request merged**
